src/modules/75-cf/pages/onboarding/views/ValidatingYourFlagView.tsx

```diff
@@ -48,20 +49,29 @@ export const ValidateYourFlagView: React.FC<TestYourFlagViewProps> = ({
     environmentIdentifier: environmentIdentifier as string
   })
 
-  const { data: featuresData, refetch } = useGetAllFeatures({
-    queryParams: {
-      accountIdentifier,
-      orgIdentifier,
+  const queryParams = useMemo<GetFeatureMetricsQueryParams>(() => {
+    return {
       projectIdentifier,
-      metrics: true,
       environmentIdentifier: environmentIdentifier as string,
-      pageSize: 1,
+      accountIdentifier,
+      orgIdentifier,
+      pageSize: CF_DEFAULT_PAGE_SIZE,
```

Comment:
Instead of bringing back 15 records, we can bring back one and limit to the specific flag's metrics using `featureIDs`
```suggestion
      pageSize: 1,
      featureIDs: [flagInfo.identifier],
```